devtools/conda-envs/test.yaml

```diff
@@ -5,7 +5,7 @@ dependencies:
   # Core
   - python
   - cython =0.29.36
-  - numpy
+  - numpy=2
```

Comment:
It'd be nice to test on version 1 and 2, injecting it in CI like we already do Python versions

Reply:
Added in #1928.